### PR TITLE
Migrate Modules Catalogue page to Symfony

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | See screenshot
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Forge?            | http://forge.prestashop.com/browse/BOGOSS-50
| How to test?  | Access `Improve > Modules > Modules Catalog`

ok

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8372)

### CODE REVIEW

File: src/PrestaShopBundle/Controller/Admin/Improve/StoreController.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink('AdminAddonsCatalog'),
+            'requireFilterStatus' => false,
+            'level' => $this->authorizationLevel($this::CONTROLLER_NAME),
+        ));
+    }
+
+    private function getAddonsUrl(Request $request)
+    {
+        $psVersion = _PS_VERSION_;
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/Improve/StoreController.php]: I don't want to to it, do you agree to declare a new PS_VERSION constant inside AppKernel and register it as container parameter? This constant will have to be updated at every release!

**Comment by Quetzacoalt91** [comment]
@mickaelandrieu The link displayed in the side menu must be updated, or we cannot reach your page.

**Comment by mickaelandrieu** [comment]
@Quetzacoalt91 you're right, let me fix it :+1:

edit => done

**Comment by eternoendless** [comment]
Reviewed 8 of 9 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[src/PrestaShopBundle/Controller/Admin/Improve/StoreController.php, line 36 at r2](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-KvbyQDgRL6iqlR7QwB1:-KvbyQDgRL6iqlR7QwB2:bjn4kwj) ([raw file](https://github.com/prestashop/prestashop/blob/7a28f1ed7d094a165eadde48c48b6d71998108b0/src/PrestaShopBundle/Controller/Admin/Improve/StoreController.php#L36)):*
> ```PHP
>  * Responsible of "Improve > Modules > Modules Catalog" page display
>  */
> class StoreController extends FrameworkBundleAdminController
> ```

Why is it called `StoreController`? :thinking: 

---

*[src/PrestaShopBundle/Controller/Admin/Improve/StoreController.php, line 59 at r2](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-Kvc6OZcr432V2jK-P_O:-Kvc6OZcr432V2jK-P_P:b-rzam0y) ([raw file](https://github.com/prestashop/prestashop/blob/7a28f1ed7d094a165eadde48c48b6d71998108b0/src/PrestaShopBundle/Controller/Admin/Improve/StoreController.php#L59)):*
> ```PHP
>     }
> 
>     private function getAddonsUrl(Request $request)
> ```

Please add Phpdoc to all new methods

---

*[src/PrestaShopBundle/Controller/Admin/Improve/StoreController.php, line 65 at r2](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-Kvc6fsKVq7VaCKPrdtX:-Kvc6fsLXJd564xwPVSD:b-d1w1fg) ([raw file](https://github.com/prestashop/prestashop/blob/7a28f1ed7d094a165eadde48c48b6d71998108b0/src/PrestaShopBundle/Controller/Admin/Improve/StoreController.php#L65)):*
> ```PHP
>         $currencyCode = $this->getContext()->currency->iso_code;
>         $languageCode = $this->getContext()->language->iso_code;
>         $countryCode = $this->getContext()->country->iso_code;
> ```

You should store the result of `$this->getContext()` in a variable in order to avoid repeated calls

---

*[src/PrestaShopBundle/Resources/config/admin/routing_module.yml, line 85 at r2](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-Kvc820LmxPdMTYM86Ya:-Kvc820LmxPdMTYM86Yb:b-n40lne) ([raw file](https://github.com/prestashop/prestashop/blob/7a28f1ed7d094a165eadde48c48b6d71998108b0/src/PrestaShopBundle/Resources/config/admin/routing_module.yml#L85)):*
> ```YAML
> 
> admin_module_store:
>     path:     /store
> ```

Why is it "store" instead of "catalog"?

---

*[src/PrestaShopBundle/Resources/views/Admin/Improve/Module/store.html.twig, line 28 at r2](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-Kvc8RJZt_LSlBgfyuQF:-Kvc8RJZt_LSlBgfyuQG:blksnls) ([raw file](https://github.com/prestashop/prestashop/blob/7a28f1ed7d094a165eadde48c48b6d71998108b0/src/PrestaShopBundle/Resources/views/Admin/Improve/Module/store.html.twig#L28)):*
> ```Twig
> 
> {% block content %}
>   {{ pageContent|raw }}
> ```

Why did you drop the iframe and make its content go through the backend?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8372)*

**Comment by mickaelandrieu** [comment]
> Why is it called StoreController? :thinking:

`catalog` is already taken in the another "Module" page :/ it's a mistake but it's too late.

> Why did you drop the iframe and make its content go through the backend?

To be honest, I don't see the value of this iframe: let's the controller do its job. The same was done in the old legacy controller, so I guess we are "iso functional" here.

I'll fix the other comments today :+1:

**Comment by eternoendless** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[src/PrestaShopBundle/Controller/Admin/Improve/StoreController.php, line 36 at r2](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-KvbyQDgRL6iqlR7QwB1:-Kw-Dll6gyeL54SxlmLe:byht09m) ([raw file](https://github.com/prestashop/prestashop/blob/7a28f1ed7d094a165eadde48c48b6d71998108b0/src/PrestaShopBundle/Controller/Admin/Improve/StoreController.php#L36)):*
> `catalog` is already taken in the another "Module" page :/ it's a mistake but it's too late.

Why not `AddonsCatalog` or `ModulesCatalog`? Some lines below there's this:

```php
    /**
     * @var string The controller name for routing.
     */
    const CONTROLLER_NAME = 'AdminAddonsCatalog';
```

---

*[src/PrestaShopBundle/Resources/views/Admin/Improve/Module/store.html.twig, line 28 at r2](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-Kvc8RJZt_LSlBgfyuQF:-Kw-Be1H_rIW8GSearym:b-98ztyf) ([raw file](https://github.com/prestashop/prestashop/blob/7a28f1ed7d094a165eadde48c48b6d71998108b0/src/PrestaShopBundle/Resources/views/Admin/Improve/Module/store.html.twig#L28)):*
> To be honest, I don't see the value of this iframe: let's the controller do its job.

There are two problems here:
1) @rgaillard raised the issue that including addons content directly could open an attack vector on all shops through addons. We need to keep the iframe.
2) Server-side requests like this create unnecessary load on the server and increase page load time, because the client won't get anything until the server has fully loaded the remote data. It can be the other way around for very specific cases (i.e. if the remote page is slow or if it's a static content than can be stored in cache) -- but I don't think it's worth it for this one. We should generally try to avoid server-side requests unless there's a specific and well-analyzed need.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8372)*

**Comment by mickaelandrieu** [comment]
> @rGaillard raised the issue that including addons content directly could open an attack vector on all shops through addons. We need to keep the iframe.

We can't "keep" it as there is no iframe at start but ok, I have no strong opinion on it and don't wan't to lose more time on a mostly static page ;)

edit: finally no, using an iframe for the same content ruins the styles.

With iframe:

testiframe

Without iframe:

noiframe

I don't wan't to lose time either at redesigning contents that comes from addons website too, so what do we do here?

**Comment by mickaelandrieu** [comment]
> Why not AddonsCatalog or ModulesCatalog? Some lines below there's this:

ModulesCatalog or AddonsCatalog creates confusion with `catalogAction` from `ModuleController`.

> const CONTROLLER_NAME = 'AdminAddonsCatalog';

this refers to the old Controller, there is nothing I can do here (or I have to update security rights on database: no gain but potential risks on Shops upgrades).

**Comment by Quetzacoalt91** [comment]
@eternoendless @mickaelandrieu, for your information this page will be moved / updated on PS 1.7.4 for a merge in the current module page. Do not spend too much time on it. :)

**Comment by eternoendless** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/PrestaShopBundle/Controller/Admin/Improve/StoreController.php, line 36 at r2](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-KvbyQDgRL6iqlR7QwB1:-KwEpqpMzpXBny3jgAKq:bx3wxe7) ([raw file](https://github.com/prestashop/prestashop/blob/7a28f1ed7d094a165eadde48c48b6d71998108b0/src/PrestaShopBundle/Controller/Admin/Improve/StoreController.php#L36)):*
> ModulesCatalog or AddonsCatalog creates confusion with catalogAction from ModuleController.

Ok, can we settle for `AddonsStore`?

---

*[src/PrestaShopBundle/Resources/views/Admin/Improve/Module/store.html.twig, line 28 at r2](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-Kvc8RJZt_LSlBgfyuQF:-KwEoC6vtyJiHPhV1hxz:bu3v3w9) ([raw file](https://github.com/prestashop/prestashop/blob/7a28f1ed7d094a165eadde48c48b6d71998108b0/src/PrestaShopBundle/Resources/views/Admin/Improve/Module/store.html.twig#L28)):*
> We can't "keep" it as there is no iframe at start

I took a better look at the old controller and it looks like it already did the server-side request, and only showed the iframe if there was no content from the remote server :thinking: ...

So if this is going to be refactored later (it will be an iframe if I'm not mistaken), we can leave the server-side request until then.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8372)*

**Comment by mickaelandrieu** [comment]
d

---

Review status: 6 of 10 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/PrestaShopBundle/Controller/Admin/Improve/AddonsStoreController.php, line 36 at r2](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-KvbyQDgRL6iqlR7QwB1:-KwFI01BNrd5LlubX-08:b8ieq3f) ([raw file](https://github.com/prestashop/prestashop/blob/7a28f1ed7d094a165eadde48c48b6d71998108b0/src/PrestaShopBundle/Controller/Admin/Improve/AddonsStoreController.php#L36)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

> ModulesCatalog or AddonsCatalog creates confusion with catalogAction from ModuleController.

Ok, can we settle for `AddonsStore`?
</blockquote></details>

Controller, route and template have been renamed: the `CONTROLLER_NAME` const value is kept to keep the associated Security rights on the page :+1: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-:-KwFHwPUyzaLoaURFdvm:b3ksf)*

**Comment by mickaelandrieu** [comment]
Review status: 6 of 10 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/PrestaShopBundle/Controller/Admin/Improve/AddonsStoreController.php, line 65 at r2](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-Kvc6fsKVq7VaCKPrdtX:-KwFIaWE0Ps4DOWL9Fkf:b-7rl0kd) ([raw file](https://github.com/prestashop/prestashop/blob/7a28f1ed7d094a165eadde48c48b6d71998108b0/src/PrestaShopBundle/Controller/Admin/Improve/AddonsStoreController.php#L65)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

You should store the result of `$this->getContext()` in a variable in order to avoid repeated calls
</blockquote></details>

done

---

*[src/PrestaShopBundle/Resources/config/admin/routing_module.yml, line 85 at r2](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-Kvc820LmxPdMTYM86Ya:-KwFIYd-0oJ1SZQhSDeL:baw55ue) ([raw file](https://github.com/prestashop/prestashop/blob/7a28f1ed7d094a165eadde48c48b6d71998108b0/src/PrestaShopBundle/Resources/config/admin/routing_module.yml#L85)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

Why is it "store" instead of "catalog"?
</blockquote></details>

done too :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8372)*

**Comment by eternoendless** [comment]
Reviewed 1 of 11 files at r1, 6 of 6 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8372)*

**Review by eternoendless — APPROVED** [review]

**Comment by eternoendless** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8372#-:-KwFgeWfaWzglUDYzR8t:bnfp4nl)*

**Comment by Quetzacoalt91** [comment]
One more! Thank you @mickaelandrieu

**Pull request merged**